framework/src/outputs/Checkpoint.C

```diff
@@ -72,6 +80,67 @@ Checkpoint::directory() const
   return _file_base + "_" + _suffix;
 }
 
+void
+Checkpoint::outputStep(const ExecFlagType & type)
+{
+  // Output is not allowed
+  if (!_allow_output && type != EXEC_FORCED)
+    return;
+
+  // If recovering disable output of initial condition, it was already output
+  if (type == EXEC_INITIAL && _app.isRecovering())
+    return;
+
+  // Check whether we should output, then do it.
+  if (shouldOutput(type))
+  {
+    TIME_SECTION("outputStep", 2, "Outputting Step");
+    output(type);
+  }
+}
+
+bool
+Checkpoint::shouldOutput(const ExecFlagType & type)
+{
+  switch (AUTOSAVE_TYPE(_is_autosave))
```

Comment:
You should be able to just say:
`switch (_is_autosave)`

It's already of an "enumerable type".